lib/urldata.h

```diff
@@ -54,13 +54,27 @@
 #define PORT_MQTT 1883
 
 #ifdef USE_WEBSOCKETS
+/* CURLPROTO_GOPHERS (29) is the highest publicly used protocol bit number,
+ * the rest are internal information. If we use higher bits we only do this on
+ * platforms that have a >= 64 bit type and then we use such a type for the
+ * protocol fields in the protocol handler.
+ */
 #define CURLPROTO_WS     (1<<30)
 #define CURLPROTO_WSS    (1LL<<31)
+
+/* This type should be bumped to a curl_off_t once we need bit 32 or higher */
+typedef unsigned int curl_prot_t;
```

Comment:
Well, we don't right now and I prefer not to work on solutions before we need them. When we get another protocol that needs >32, then we adjust this to depend on >32 bit protocol numbers. Possibly then we have some ifdef logic somewhere that sets a new define name we can use for this.

Reply:
I don't understand your response, sorry.
`curl_prot_t` is defined with exactly the same way in both the true and false branches of the #ifdef.
I agree we can keep it `unsigned int` yet. But if conditioned, it should not be on `USE_WEBSOCKET`.

Reply:
We don't have any other protocol than websockets that triggers this condition so we have not yet created any "SUPPORTS_32BITPLUS_PROTOCOLS" define. That's why.

It is the same in both because it's not necessary to be larger until we add the next bit. As the comment explains.

Back to: don't implement functionality until needed.

Reply:
Ah this is a transitional situation. This is much clearer after your explanation.

Reply:
Well, everything is always transitional in one way or another....

Reply:
I first did a test build with a larger type to make sure it works, and I worked out the nits for that before I set the smaller size even for websockets.